pkg/injector/config_test.go

```diff
@@ -28,7 +28,7 @@ func TestGetInjectorConfig(t *testing.T) {
 		t.Setenv("SIDECAR_IMAGE_PULL_POLICY", "Always")
 		t.Setenv("NAMESPACE", "test-namespace")
 		t.Setenv("KUBE_CLUSTER_DOMAIN", "cluster.local")
-		t.Setenv("ALLOWED_SERVICE_ACCOUNTS", "test-service-account1:test1,test-service-account2:test2")
+		t.Setenv("ALLOWED_SERVICE_ACCOUNTS", "test1:test-service-account1,test2:test-service-account2")
```

Comment:
Although this makes a lot of sense to me, it's also a breaking change.

Perhaps we should keep `ALLOWED_SERVICE_ACCOUNTS` behaving as it does today, mark it as deprecated (using it would show a warning log, but would still work), and introduce a new env var (for example `ALLOWED_SA` ? not sure about the name) that uses the new format

Reply:
https://github.com/dapr/dapr/pull/5892#issuecomment-1419379947.